🐍 🛤️ Python result tracker

mberr opened the pull request:
This PR adds a simple in-memory result tracker, which stores parameters and metrics in Python dictionaries.

mberr committed:
add python result tracker

mberr committed:
add test

trigger ci

mberr committed:
add type annotation

trigger ci

mberr committed:
remove unused import

trigger ci

mberr committed:
update readme

trigger ci

mberr committed:
fix types

trigger ci

cthoyt commented on src/pykeen/trackers/base.py:
```diff
@@ -48,6 +49,44 @@ def end_run(self, success: bool = True) -> None:
         """
 
 
+class PythonResultTracker(ResultTracker):
```
documentation request: how would one go about accessing the result tracker after training with the pipeline?

mberr replied:
6a06ed0

cthoyt replied:
thanks :)

mberr committed:
export PythonResultTracker

mberr committed:
use Resolver.from_subclasses

mberr committed:
fix import

mberr committed:
add example use case

trigger ci

cthoyt reviewed (approved):
:shipit:

mberr committed:
trigger ci

mberr committed:
remove unused import

trigger ci

cthoyt committed:
Update base.py

Trigger CI

@mberr be careful on the placing of the spaces with code blocks - you can always try building the docs locally to check that they render properly

mberr committed:
update docstring

trigger ci

cthoyt commented on src/pykeen/trackers/base.py:
```diff
@@ -50,8 +50,7 @@ def end_run(self, success: bool = True) -> None:
 class PythonResultTracker(ResultTracker):
-    """
-    A tracker which stores everything in Python dictionaries.
+    """A tracker which stores everything in Python dictionaries.
```
😮

mberr replied:
`pykeen readme` does not like empty first lines 😕

cthoyt replied:
Haha this is obviously my doing. We could add a string strip in there so you can continue to write docstrings this way

mberr committed:
Merge remote-tracking branch 'origin/master' into python-result-tracker

mberr committed:
trigger ci
